pkg/antctl/raw/featuregates/command_test.go

```diff
@@ -0,0 +1,400 @@
+			getFakeRestClient: func(kubeconfig *rest.Config, k8sClientset kubernetes.Interface, antreaClientset antrea.Interface, mode string) (*rest.RESTClient, error) {
+				fakeHttpClient := fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+					return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(bytes.NewBuffer(agentResponse))}, nil
+				})
+				restClient.Client = fakeHttpClient
+				return restClient, nil
+			},
+		},
+		{
+			name:           "get featuregates in controller Pod",
+			runE:           controllerLocalRunE,
+			expectedOutput: "Antrea Agent Feature Gates\nFEATUREGATE              STATUS         VERSION   \nEndpointSlice            Disabled       ALPHA     \nAntreaIPAM               Disabled       ALPHA     \nFlowExporter             Disabled       ALPHA     \nNetworkPolicyStats       Enabled        BETA      \nAntreaPolicy             Enabled        BETA      \nMulticast                Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nAntreaProxy              Enabled        BETA      \nEgress                   Enabled        BETA      \nNodePortLocal            Enabled        BETA      \nMulticluster             Enabled        ALPHA     \nAntrea Controller Feature Gates\nFEATUREGATE              STATUS         VERSION   \nAntreaPolicy             Enabled        BETA      \nNodeIPAM                 Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nEgress                   Enabled        BETA      \nNetworkPolicyStats       Enabled        BETA      \n",
+			getFakeRestClient: func(kubeconfig *rest.Config, k8sClientset kubernetes.Interface, antreaClientset antrea.Interface, mode string) (*rest.RESTClient, error) {
+				fakeHttpClient := fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+					return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(bytes.NewBuffer(fullResponse))}, nil
+				})
+				restClient.Client = fakeHttpClient
+				return restClient, nil
+			}},
+	}
+
+	getClients = func(cmd *cobra.Command) (*rest.Config, kubernetes.Interface, antrea.Interface, error) {
+		return clientConfig, k8sClient, antreaClientset, nil
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			getRestClient = tt.getFakeRestClient
+			buf := new(bytes.Buffer)
+			Command.SetOutput(buf)
+			Command.SetOut(buf)
+			Command.SetErr(buf)
+
+			err := tt.runE(Command, nil)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expectedOutput, buf.String())
+		})
+	}
+}
+
+func TestGetConfigAndClients(t *testing.T) {
```

Comment:
I'm still not convinced that these tests (`TestGetConfigAndClients` and `TestGetRestClientByMode`) are useful, but I'll let others comment as well.

Reply:
`TestGetConfigAndClients` helped to find out the issue where the `server` option actually doesn't work in old codes.
`TestGetRestClientByMode` is to improve the unit test coverage which not covered by other test cases.
I guess it's better to keep it.